From 4b9756060f994b2aa6be36403108d4ceed4942da Mon Sep 17 00:00:00 2001
From: Mahesh Mahadevan <r9aadq@freescale.com>
Date: Fri, 13 Jun 2008 10:53:41 -0500
Subject: [PATCH] ENGR00080371: Disable I-Cache along with MMU after decompressing the kernel

The I-cache is diabled after kernel decompression when the MMU is disabled

Signed-off-by: Mahesh Mahadevan <r9aadq@freescale.com>
---
 arch/arm/boot/compressed/head.S |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/compressed/head.S b/arch/arm/boot/compressed/head.S
index 5cac46a..b605417 100644
--- a/arch/arm/boot/compressed/head.S
+++ b/arch/arm/boot/compressed/head.S
@@ -705,6 +705,9 @@ __armv4_mmu_cache_off:
 __armv7_mmu_cache_off:
 		mrc	p15, 0, r0, c1, c0
 		bic	r0, r0, #0x000d
+#if defined(CONFIG_ARCH_MX51)
+		bic	r0, r0, #0x1000
+#endif
 		mcr	p15, 0, r0, c1, c0	@ turn MMU and cache off
 		mov	r12, lr
 		bl	__armv7_mmu_cache_flush
-- 
1.5.4.4

